crates/ursa-network/src/tests/service_tests.rs

```diff
@@ -61,21 +61,20 @@ mod tests {
         }
     }
 
-    async fn _run_bootstrap(
+    async fn run_bootstrap(
         config: &mut NetworkConfig,
-        port: u16,
-    ) -> Result<(UrsaService<MemoryDB>, String)> {
+    ) -> Result<(UrsaService<MemoryDB>, Multiaddr, PeerId)> {
         let keypair = Keypair::generate_ed25519();
-        let swarm_addr = format!("/ip4/127.0.0.1/tcp/{port}");
-        config.swarm_addrs = vec![swarm_addr.clone().parse().unwrap()];
-        let addr = format!("{}/p2p/{}", swarm_addr, PeerId::from(keypair.public()));
-        let (bootstrap, ..) = network_init(config, Some(addr.clone()), Some(keypair)).await?;
-        Ok((bootstrap, addr))
+        config.swarm_addrs = vec!["/ip4/127.0.0.1/tcp/0".parse().unwrap()];
+        config.bootstrapper = true;
+        config.bootstrap_nodes = vec![];
+        let (bootstrap, addr, peer_id, ..) = network_init(config, None, Some(keypair)).await?;
+        Ok((bootstrap, addr, peer_id))
     }
 
     async fn network_init(
```

Comment:
we don't need to start two nodes for this test; one is fine to test that the network inits